### PR TITLE
Adding initial NX support

### PR DESCRIPTION
**Pull request by apinkert**
Moving into code review to get some eyes while getting the release stage in GH actions working

### CODE REVIEW

**Comment by Hyperkid123** [comment]
About the release job.

First, there needs to be a couple of tags so the scripts have some base. I can give you more details in PMs. The release job should be also disabled on PRs

There also seem to be quite a lot of logging happening caused by the maven installation. Is there a way to "silence them"? Its going to be a pain to find errors in the logs when some occur.

File: packages/vulnerabilities/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@redhat-cloud-services/vulnerabilities-client",
-  "version": "1.2.13",
+  "version": "1.2.14",
```
Comment (Hyperkid123) [inline packages/vulnerabilities/package.json:3]: Last thing before release. We will have to make sure, all the tags, releases and actual npm package versions are in sync. For example this version does not exist on npm:

https://www.npmjs.com/package/@redhat-cloud-services/vulnerabilities-client?activeTab=versions

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -57,7 +57,10 @@ jobs:
       - uses: actions/checkout@v3
         with:
           fetch-depth: 0
+          ssh-key: ${{ secrets.GH_BOT_TOKEN }}
```
Comment (Hyperkid123) [inline .github/workflows/ci.yml:60]: I can't really inspect the secret, but my gut tells me its not an SSH key of our bot.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -57,7 +57,10 @@ jobs:
...
       - uses: './.github/actions/release'
         with:
           gh_token: ${{ secrets.GITHUB_TOKEN }}
-          gh_actor: ${GITHUB_ACTOR}
\ No newline at end of file
+          npm_token: ${{ secrets.NPM_TOKEN }}
```
Comment (Hyperkid123) [inline .github/workflows/ci.yml:64]: We do not have this secret configured right now. I'll add it to the secrets.

**Comment by apinkert** [comment]
For an update on this PR, currently waiting for a ticket to create an email for nacho-bot to be used to sign commits/potentially publish npm packages for us. Once that final secret gets setup for the release GH action, I'll clean up this PR and squash all the test version commits

**Review by Hyperkid123 — APPROVED** [review]

File: .commonlintrc.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": ["@commitlint/config-conventional"],
+  "rules": {
+    "body-max-line-length": [0]
+  }
+}
```
[the remark this answers is not in the document]
Comment (Hyperkid123) [reply .commonlintrc.json:6]: I think we are simply not linting the JSON files.

**Pull request merged**